tests/ignite/handlers/test_time_limit.py

```diff
@@ -0,0 +1,43 @@
+import time
+
+import pytest
+
+from ignite.engine import Engine, Events
+from ignite.handlers import TimeLimit
+
+
+def test_arg_validation():
+
+    with pytest.raises(ValueError, match=r"Argument limit_sec should be a positive integer."):
+        TimeLimit(limit_sec=-5)
+
+    with pytest.raises(TypeError, match=r"Argument limit_sec should be an integer."):
+        TimeLimit(limit_sec="abc")
+
+
+def test_terminate_on_time_limit():
```

Comment:
maybe we could also test the handler by the overall time spent, like should not be larger expected time + 1 minute .. ?

Reply:
>  like should not be larger expected time + 1 minute .. ?

you mean testing the internal state of TimeLimit right?

Reply:
I mean something like that : 
```python

started = time.time()

expected_time = 5
trainer = ...
handler = TimeLimit(expected_time)
trainer.add_event_handler(Events.ITERATION_COMPLETED, handler)
trainer.run(...)

elapsed = time.time() - started
assert elapsed + 1 < expected_time
```
what do you think ?